src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -623,6 +637,26 @@ private SecurityGroupApi getSecurityGroupApi(Infrastructure infrastructure) {
         }
     }
 
+    private AWSSecurityGroupApi getAWSSecurityGroupApi(Infrastructure infrastructure) {
+        ComputeService computeService = getComputeServiceFromInfastructure(infrastructure);
+        AWSEC2Api awsEc2Api = computeService.getContext().unwrapApi(AWSEC2Api.class);
+        if (awsEc2Api.getSecurityGroupApi().isPresent()) {
+            return awsEc2Api.getSecurityGroupApi().get();
+        } else {
+            throw new UnsupportedOperationException("Cannot retrieve AWS security group API");
+        }
+    }
+
+    private AWSSubnetApi getAWSSubnetApi(Infrastructure infrastructure) {
+        ComputeService computeService = getComputeServiceFromInfastructure(infrastructure);
+        AWSEC2Api awsEc2Api = computeService.getContext().unwrapApi(AWSEC2Api.class);
+        if (awsEc2Api.getAWSSubnetApi().isPresent()) {
+            return awsEc2Api.getAWSSubnetApi().get();
+        } else {
+            throw new UnsupportedOperationException("Cannot retrieve AWS subnet API");
```

Comment:
can you clarify more when this would happen?

Reply:
Same